### PR TITLE
Add mode, branch, API env, OneSky, and more setup features

### PR DESCRIPTION
**Pull request by OzzieOrca**
- Cleanup setup code
- Add brew install script
- Add repo cloning
- Add Android AVD creation
- Add OneSky keys setup and download step
- Add mode selection
- Add git branch selection
- Add API env selection

I had fun. Hopefully it wasn't too much of a rewrite :) I welcome any feedback.

How are you sharing this with the testers? Did you publish it somewhere? (If they have node all they would need is `npx mh-qa-cli` or whatever.) Or you just manually installing it?

I didn't figure out how to make Android start with a skin (Pixel 3 XL, etc). I could try some more. Also, it may just not work depending on what Android tools and API versions they have installed. Also the Android emulator keeps the CLI open. I tried getting it to spawn and not die with the parent but I didn't figure it out.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,185 +1,222 @@
...
 }
 
-async function ls() {
-  const { stdout, stderr } = await exec('ls');
-  console.log('stdout:', stdout);
-  console.log('stderr:', stderr);
+async function fetchBranches() {
+  cli.action.start('👀  Fetching branches');
+
+  await exec('cd ~/code/missionhub-react-native && git fetch origin --prune');
```
Comment (soberstadt) [inline src/index.ts]: You don't need this if below you use
```
"cd ~/code/missionhub-react-native && git ls-remote -q --heads | awk '{print $2}'"
```

Comment (OzzieOrca) [reply src/index.ts]: Thanks ya I'm not the best with git/bash commands.

**Comment by OzzieOrca** [comment]
Can you elaborate on
> Could you add some defaults to the inquirer?

The first item should be default and I've sorted them by what made sense to me. Or were you talking about the command line args?

**Comment by soberstadt** [comment]
I haven't actually tested it, so your answer of it being the first one make sense.

File: src/index.ts [hunk 2]
```diff
@@ -112,22 +112,29 @@ async function askForBranch() {
...
 
-  await exec('cd ~/code/missionhub-react-native && git fetch origin --prune');
   const { stdout } = await exec(
-    'cd ~/code/missionhub-react-native && git branch -r',
+    "cd ~/code/missionhub-react-native && git ls-remote -q --heads | awk '{print $2}'",
   );
 
-  const branches = stdout
+  const sortPriority = (branch: string) =>
+    branch === 'develop'
```
Comment (soberstadt) [inline src/index.ts]: Now this ternary i can understand and get behind. The one you took out was hard to mentally parse.

Comment (OzzieOrca) [reply src/index.ts]: Ya it wasn't sorting right. I just needed to take another look at it...

**Pull request merged**